### PR TITLE
Ele 3022 dbt 180 fixes

### PR DESCRIPTION
**Pull request by haritamar**

### CODE REVIEW

File: integration_tests/dbt_project/macros/materializations.sql [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+{% endmaterialization %}
+
+{% materialization incremental, adapter="trino", supported_languages=["sql"] %}
+  {% do return(elementary.materialization_incremental_trino()) %}
+{% endmaterialization %}
+
+{# Test #}
+
+{% materialization test, default %}
+  {% if var('enable_elementary_test_materialization', false) %}
```
Comment (haritamar) [inline integration_tests/dbt_project/macros/materializations.sql:68]: Wanted the integration tests to run by default without the test materialization, so we can catch exactly the cases that require it to work properly.
Thought to do the same for the other materializations but it was a bit of a hassle so I didn't.

File: macros/edr/tests/on_run_start/ensure_materialize_override.sql [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        {% do elementary.file_log("Materialization override is enabled.") %}
+        {% do return(none) %}
+    {% endif %}
+
+    {% set major, minor, revision = dbt_version.split(".") %}
+    {% set major = major | int %}
+    {% set minor = minor | int %}
+    {% if major > 1 or major == 1 and minor >= 8 %}
+        {%- set msg %}
+IMPORTANT - Starting from dbt 1.8, users must explicitly allow packages to override materializations. Elementary requires this ability to support collection of samples and failed row count for dbt tests.
```
Comment (elongl) [inline macros/edr/tests/on_run_start/ensure_materialize_override.sql:17]: Maybe drop line before `Elementary requires...`?

File: macros/utils/graph/get_run_result_dict.sql [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{% macro get_run_result_dict(run_result) %}
+    {% if dbt_version <= "1.8" %}
```
Comment (elongl) [inline macros/utils/graph/get_run_result_dict.sql:2]: Fix version check here.

**Review by elongl — APPROVED** [review]

File: .github/workflows/test-all-warehouses.yml [hunk 4]
```diff
@@ -37,8 +37,8 @@ jobs:
       matrix:
         dbt-version:
           ${{ inputs.dbt-version && fromJSON(format('["{0}"]', inputs.dbt-version)) ||
-          ! contains(github.event_name, 'pull_request') && fromJSON('["1.3.0", "latest_official"]') ||
-          fromJSON('["latest_official"]') }}
+          ! contains(github.event_name, 'pull_request') && fromJSON('["1.3.0", "latest_pre"]') ||
+          fromJSON('["latest_pre"]') }}
```
[the remark this answers is not in the document]
Comment (haritamar) [reply .github/workflows/test-all-warehouses.yml:41]: We decided not to for now

**Pull request merged**